cpp/include/cudf/column/column_view.hpp

```diff
@@ -289,7 +289,7 @@ class column_view_base {
                    size_type size,
                    void const* data,
                    bitmask_type const* null_mask = nullptr,
-                   size_type null_count          = UNKNOWN_NULL_COUNT,
+                   size_type null_count          = 0,
```

Comment:
Do we have any validity check for null count? If we specify some `null_mask`, but leave `null_count` default to `0`, will the column_view be constructed as-is?

Reply:
Answer to myself: We don't have any validity check. So in such cases, we will have invalid `null_count`. I would suggest to add such check, which mean we have to execute the kernel to count nulls upon column_view construction.

Reply:
That's a good point. With this API change, it becomes easier to create a column with nulls and zero null count. Ideally we would not have a default `null_count` when `null_mask` is passed.

Reply:
One way to do this would be to tie the two parameters
```
struct null_mask{
  bitmask_type const* mask;
  size_type count;
};
```
instead of passing these dependent parameters separately. But something like this would be a **very** breaking change.

Reply:
That sounds reasonable. We can make such breaking change at the end after completely removing `UNKNOWN_NULL_COUNT`.

Reply:
I am thinking of removing the default value for both null mask and null count to avoid potential bug. It will not be breaking changes, as the callers can just pass in an empty null mask. How do you guys think about it?

Reply:
I had discussed this recently with @davidwendt and thought that @ttnghia had actually already proposed what @vuule did above. Looking through recent issues, I suspect he was referring to #13154. That issue is a subset of what David and I had discussed, which was indeed redefining the null mask everywhere as a pair containing both pieces of information. That would enforce that you could never read or write one without the other. Whether we actually want to go quite that far, or just modify select APIs, remains to be seen. I definitely think that is the right direction for us to take though.

@vuule @ttnghia what do you think is a minimal acceptable change for:
1. Moving this PR forward
2. The 23.06 release
3. The long-term API changes

Reply:
This PR can potentially break user code, when they create a column with nulls and default null count.
We need an API that takes both mask and count, or none. AFAICT, we can't get to that API without an API break. Which means we can't remove `UNKNOWN_NULL_COUNT` without an API break.

I'm not too experienced with non-cuIO API changes, but it seems like we'd have to deprecate these functions and offer an alternative that's safe to use in libcudf that has no `UNKNOWN_NULL_COUNT`. Then we can remove the old functions in the next release.

Reply:
I'm fine with moving merging PR as-is to avoid more friction, assuming that we will address all our concerns above in the follow up PR. Merging this as-is will open a loophole for bug, so we will need to close it by the follow up (breaking change) work ASAP.

Reply:
libcudf doesn't promise deprecations, we only say we'll make a best effort. In this instance I think that ship has already failed since David and I have had at least one API-breaking change merged. Our hope at this stage is to get all the breaking changes that we want around `UNKNOWN_NULL_COUNT` merged into this release. So I'm inclined to follow Nghia's recommendation of merging this PR ASAP, then making follow-ups immediately to address the outstanding issues.

Reply:
What is the plan for the follow-up changes?

Reply:
After this and #13341 are merged, we'll be ready to remove `UNKNOWN_NULL_COUNT`. Then we have two choices for the immediate follow-up:
- @vuule's suggestion: Couple the two parameters (mask and count) for some set of APIs by changing to require a pair as input
- @ttnghia's suggestion: Remove the default null mask parameter
and I suppose there is the question of whether we think this is a must for 23.06.

@davidwendt what do you think?

Reply:
Unless we have other potential uses for a `null_mask_with_known_null_count`, it's probably better to just remove the default parameters.

Reply:
I thought that changing the signature could be something we would do in a future release. I think I'm convinced that changing to remove the default parameters or introducing the pair type would be more responsible. I  suspect removing the defaults may be the easier path.

Reply:
I was able to remove the default parameter without too much fuss. The total diff is still fairly small, so I think that was the right call to do in this PR.

Reply:
@vuule @ttnghia does that change look good to you, or is there anything else you'd like to see in this PR?

Reply:
Yes, that's good. Thanks :+1:

Reply:
Please update the PR description as well :)

Reply:
Good call. Updated the description and the PR title.